### ISSUE TITLE
Implement currentColor in SVG

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#556]
It would be useful to stroke elements in SVG images with `currentColor` rather than `#000000`.

Info about currentColor:

https://developer.mozilla.org/en/docs/Web/CSS/color_value#currentcolor_keyword
https://css-tricks.com/cascading-svg-fill-color

Example CSS:

```css
<style>
        svg g.note {
                color: red;
                stroke: deeppink !important;
                fill: blue;
        }
</style>
```

Current SVG output from verovio uses a stroke of `#000000` for stems:

```xml
      <g class="note" id="note-L2F1">
        <use xlink:href="#E0A4" x="1061" y="1940" height="720px" width="720px" />
        <g class="stem" id="stem-1">
         <path d="M1277 1918 L1277 1310" stroke="#000000" stroke-width="20" />
         <g class="flag" id="flag-1">
          <use xlink:href="#E240" x="1267" y="1310" height="720px" width="720px" />
         </g>
        </g>
```

This will be rendered as:

<img width="1349" alt="screen shot 2017-04-22 at 8 19 41 am" src="https://cloud.githubusercontent.com/assets/3487289/25305661/7664fdb0-2734-11e7-9597-5f2a286d596d.png">

The stem remains black, even if `!important` is added to try to override the black color:
https://css-tricks.com/when-using-important-is-the-right-choice

However, if the stroke is set to `currentColor`, CSS styling can be applied:

```xml
      <g class="note" id="note-L2F1">
        <use xlink:href="#E0A4" x="1061" y="1940" height="720px" width="720px" />
        <g class="stem" id="stem-1">
         <path d="M1277 1918 L1277 1310" stroke="currentColor" stroke-width="20" />
         <g class="flag" id="flag-1">
          <use xlink:href="#E240" x="1267" y="1310" height="720px" width="720px" />
         </g>
        </g>
```

which is rendered as:

<img width="1335" alt="screen shot 2017-04-22 at 8 21 54 am" src="https://cloud.githubusercontent.com/assets/3487289/25305683/c458d140-2734-11e7-98f6-61e750f89c06.png">

In this case the stem is colored red from the CSS `color` parameter setting (not `stroke` since that is hard-coded to the soft variable `currentColor`).

Also if `fill="#000000` is ever used (it is not in my test image), then it should be replaced with `fill=currentColor` unless black is explicitly wanted such as from an MEI `color="#000000"` parameter.

Also note that the notehead should be placed after the stem element in SVG to make the stem go behind the notehead (I think that is already a reported issue).

In other words:
```xml
       <g class="note" id="note-L2F1">
        <g class="stem" id="stem-0000000768152788">
         <path d="M1277 1918 L1277 1310" stroke="currentColor" stroke-width="20" />
         <g class="flag" id="flag-0000001819705799">
          <use xlink:href="#E240" x="1267" y="1310" height="720px" width="720px" />
         </g>
        </g>
        <use xlink:href="#E0A4" x="1061" y="1940" height="720px" width="720px" />
       </g>
```

Which renders as:

<img width="1350" alt="screen shot 2017-04-22 at 8 25 56 am" src="https://cloud.githubusercontent.com/assets/3487289/25305729/66783b6e-2735-11e7-8992-1935adf85a67.png">

**Comment by rettinghaus** [comment]
Basically a good idea. But (and that's kind of a big BUT) 
Browsers treat `currentColor` like the default Color which is `#000` 
Other progs like the GIMP don't and make it transparent. If you import it into GIMP you'll get something like: 

![whitespace_001](https://cloud.githubusercontent.com/assets/7693447/25307049/c6664948-279a-11e7-880c-d3b64636d219.png)

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="792px" height="373px" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" overflow="visible">
	<desc>Engraved by Verovio 0.9.14-dev-beefb06-dirty</desc>
	<defs>
		<symbol id="E050" viewBox="0 0 1000 1000" overflow="inherit">
			<path transform="scale(1,-1)" d="M441 -245c-23 -4 -48 -6 -76 -6c-59 0 -102 7 -130 20c-88 42 -150 93 -187 154c-26 44 -43 103 -48 176c-4 60 11 123 44 189c29 57 65 106 110 148s96 85 153 127c-3 16 -8 46 -13 92c-4 43 -5 73 -5 89c0 117 16 172 69 257c34 54 64 82 89 82c21 0 43 -30 69 -92 s39 -115 41 -159c2 -120 -19 -173 -67 -256c-13 -20 -63 -90 -98 -118c-13 -9 -25 -19 -37 -29l31 -181c8 1 18 2 28 2c58 0 102 -12 133 -35c59 -43 92 -104 98 -184c11 -135 -80 -229 -180 -270c8 -57 17 -110 25 -162c5 -31 6 -58 6 -80c0 -30 -5 -53 -14 -70 c-35 -64 -88 -99 -158 -103c-42 -3 -83 6 -124 26c-50 24 -77 59 -80 105c-2 34 5 63 20 87c18 28 45 42 79 44c51 4 99 -40 103 -87c4 -56 -30 -94 -105 -115c17 -24 51 -36 102 -36c62 0 116 43 140 85c9 16 13 41 13 74c0 20 -1 42 -5 67c-8 53 -18 106 -26 159zM461 939 c-95 0 -135 -175 -135 -286c0 -24 2 -48 5 -71c50 39 92 82 127 128c43 57 63 106 60 148c-4 54 -23 82 -57 81zM406 119l54 -326c80 27 116 88 109 184c-7 99 -62 146 -163 142zM382 117c-74 -2 -132 -50 -128 -127c2 -46 43 -99 75 -115c-3 -2 -7 -5 -10 -10 c-70 33 -116 88 -123 172c-5 73 42 135 88 170c23 17 49 29 78 36l-29 170c-21 -13 -52 -37 -92 -73c-50 -44 -86 -84 -109 -119c-49 -75 -71 -140 -67 -195c5 -68 35 -127 93 -176s125 -73 203 -73c25 0 50 3 75 9c-19 111 -36 221 -54 331z" />
		</symbol>
		<symbol id="E084" viewBox="0 0 1000 1000" overflow="inherit">
			<path transform="scale(1,-1)" d="M0 -78c84 97 114 180 134 329h170c-13 -32 -82 -132 -99 -151l-84 -97c-33 -36 -59 -63 -80 -81h162v102l127 123v-225h57v-39h-57v-34c0 -43 19 -65 57 -65v-34h-244v36c48 0 60 26 60 70v27h-203v39z" />
		</symbol>
		<symbol id="E0A2" viewBox="0 0 1000 1000" overflow="inherit">
			<path transform="scale(1,-1)" d="M198 133c102 0 207 -45 207 -133c0 -92 -98 -133 -207 -133c-101 0 -198 46 -198 133c0 88 93 133 198 133zM293 -21c0 14 -3 29 -8 44c-7 20 -18 38 -33 54c-20 21 -43 31 -68 31l-20 -2c-15 -5 -27 -14 -36 -28c-4 -9 -6 -17 -8 -24s-3 -16 -3 -27c0 -15 3 -34 9 -57 s18 -41 34 -55c15 -15 36 -23 62 -23c4 0 10 1 18 2c19 5 32 15 40 30s13 34 13 55z" />
		</symbol>
	</defs>
	<style type="text/css">g.page-margin{font-family:Times;} g.tempo{font-weight:bold;} g.dir, g.dynam {font-style:italic;}</style>
	<svg id="definition-scale" viewBox="0 0 7920 3730">
		<g class="page-margin" transform="translate(500, 500)">
			<g class="system" id="system-0000001459809759">
				<g class="section boundaryStart" id="section-0000001175395249" />
				<g class="measure" id="measure-0000000165881190">
					<g class="staff" id="staff-0000000349825458">
						<path d="M0 1612 L3043 1612" stroke="currentColor" stroke-width="20" />
						<path d="M0 1792 L3043 1792" stroke="currentColor" stroke-width="20" />
						<path d="M0 1972 L3043 1972" stroke="currentColor" stroke-width="20" />
						<path d="M0 2152 L3043 2152" stroke="currentColor" stroke-width="20" />
						<path d="M0 2332 L3043 2332" stroke="currentColor" stroke-width="20" />
						<g class="clef" id="clef-0000000546677535">
							<use xlink:href="#E050" x="90" y="2152" height="720px" width="720px" />
						</g>
						<g class="meterSig" id="msig-0000002084050317">
							<use xlink:href="#E084" x="735" y="1792" height="720px" width="720px" />
							<use xlink:href="#E084" x="735" y="2152" height="720px" width="720px" />
						</g>
						<g class="ledgerLines below">
							<path d="M1233 2512 L1623 2512" stroke="currentColor" stroke-width="35" />
						</g>
						<g class="layer" id="layer-0000001853730767">
							<g class="note" id="note-0000002107733940">
								<use xlink:href="#E0A2" x="1283" y="2512" height="720px" width="720px" />
							</g>
						</g>
					</g>
					<g class="dir" id="dir-0000000991882040">
						<text x="1428" y="1479" font-size="0px">
							<tspan font-size="405px" class="text">Allegro </tspan>
							<tspan class="rend" id="rend-0000001375531017">
								<tspan font-size="405px" class="text">molto</tspan>
							</tspan>
						</text>
					</g>
					<g class="barLineAttr" id="bline-0000000178519931">
						<path d="M3028 1612 L3028 2332" stroke="currentColor" stroke-width="30" />
					</g>
				</g>
				<g class="measure" id="measure-0000001070520604">
					<g class="staff" id="staff-0000001441475186">
						<path d="M3043 1612 L4983 1612" stroke="currentColor" stroke-width="20" />
						<path d="M3043 1792 L4983 1792" stroke="currentColor" stroke-width="20" />
						<path d="M3043 1972 L4983 1972" stroke="currentColor" stroke-width="20" />
						<path d="M3043 2152 L4983 2152" stroke="currentColor" stroke-width="20" />
						<path d="M3043 2332 L4983 2332" stroke="currentColor" stroke-width="20" />
						<g class="layer" id="layer-0000001110429295">
							<g class="note" id="note-0000001352268635">
								<use xlink:href="#E0A2" x="3223" y="2422" height="720px" width="720px" />
							</g>
						</g>
					</g>
					<g class="dir" id="dir-0000000479487140">
						<text x="3368" y="1150" font-size="0px">
							<tspan class="rend" id="rend-0000001381718436" xml:space="preserve">
								<tspan font-size="405px" class="text">a         b       c</tspan>
							</tspan>
						</text>
					</g>
					<g class="barLineAttr" id="bline-0000001580895410">
						<path d="M4968 1612 L4968 2332" stroke="currentColor" stroke-width="30" />
					</g>
				</g>
				<g class="measure" id="measure-0000001950076222">
					<g class="staff" id="staff-0000001529093010">
						<path d="M4983 1612 L6923 1612" stroke="currentColor" stroke-width="20" />
						<path d="M4983 1792 L6923 1792" stroke="currentColor" stroke-width="20" />
						<path d="M4983 1972 L6923 1972" stroke="currentColor" stroke-width="20" />
						<path d="M4983 2152 L6923 2152" stroke="currentColor" stroke-width="20" />
						<path d="M4983 2332 L6923 2332" stroke="currentColor" stroke-width="20" />
						<g class="layer" id="layer-0000000529415421">
							<g class="note" id="note-0000000860231226">
								<use xlink:href="#E0A2" x="5163" y="2332" height="720px" width="720px" />
							</g>
						</g>
					</g>
					<g class="dir" id="dir-0000001631495210">
						<text x="5308" y="822" font-size="0px">
							<tspan class="rend" id="rend-0000001468789574" xml:space="default">
								<tspan font-size="405px" class="text">a      b      c</tspan>
							</tspan>
						</text>
					</g>
					<g class="barLineAttr" id="bline-0000000686488442">
						<path d="M6908 1612 L6908 2332" stroke="currentColor" stroke-width="30" />
					</g>
				</g>
				<g class="boundaryEnd section-0000001175395249" id="bdend-0000000628458526" />
			</g>
		</g>
	</svg>
</svg>
```

**Comment by rettinghaus** [comment]
or: 
![whitespace_001](https://cloud.githubusercontent.com/assets/7693447/25307106/a204e77a-279b-11e7-9b3c-8fae34913649.png)
```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="792px" height="373px" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" overflow="visible">
	<desc>Engraved by Verovio 0.9.14-dev-beefb06-dirty</desc>
	<defs>
		<symbol id="E050" viewBox="0 0 1000 1000" overflow="inherit">
			<path transform="scale(1,-1)" d="M441 -245c-23 -4 -48 -6 -76 -6c-59 0 -102 7 -130 20c-88 42 -150 93 -187 154c-26 44 -43 103 -48 176c-4 60 11 123 44 189c29 57 65 106 110 148s96 85 153 127c-3 16 -8 46 -13 92c-4 43 -5 73 -5 89c0 117 16 172 69 257c34 54 64 82 89 82c21 0 43 -30 69 -92 s39 -115 41 -159c2 -120 -19 -173 -67 -256c-13 -20 -63 -90 -98 -118c-13 -9 -25 -19 -37 -29l31 -181c8 1 18 2 28 2c58 0 102 -12 133 -35c59 -43 92 -104 98 -184c11 -135 -80 -229 -180 -270c8 -57 17 -110 25 -162c5 -31 6 -58 6 -80c0 -30 -5 -53 -14 -70 c-35 -64 -88 -99 -158 -103c-42 -3 -83 6 -124 26c-50 24 -77 59 -80 105c-2 34 5 63 20 87c18 28 45 42 79 44c51 4 99 -40 103 -87c4 -56 -30 -94 -105 -115c17 -24 51 -36 102 -36c62 0 116 43 140 85c9 16 13 41 13 74c0 20 -1 42 -5 67c-8 53 -18 106 -26 159zM461 939 c-95 0 -135 -175 -135 -286c0 -24 2 -48 5 -71c50 39 92 82 127 128c43 57 63 106 60 148c-4 54 -23 82 -57 81zM406 119l54 -326c80 27 116 88 109 184c-7 99 -62 146 -163 142zM382 117c-74 -2 -132 -50 -128 -127c2 -46 43 -99 75 -115c-3 -2 -7 -5 -10 -10 c-70 33 -116 88 -123 172c-5 73 42 135 88 170c23 17 49 29 78 36l-29 170c-21 -13 -52 -37 -92 -73c-50 -44 -86 -84 -109 -119c-49 -75 -71 -140 -67 -195c5 -68 35 -127 93 -176s125 -73 203 -73c25 0 50 3 75 9c-19 111 -36 221 -54 331z" />
		</symbol>
		<symbol id="E084" viewBox="0 0 1000 1000" overflow="inherit">
			<path transform="scale(1,-1)" d="M0 -78c84 97 114 180 134 329h170c-13 -32 -82 -132 -99 -151l-84 -97c-33 -36 -59 -63 -80 -81h162v102l127 123v-225h57v-39h-57v-34c0 -43 19 -65 57 -65v-34h-244v36c48 0 60 26 60 70v27h-203v39z" />
		</symbol>
		<symbol id="E0A2" viewBox="0 0 1000 1000" overflow="inherit">
			<path transform="scale(1,-1)" d="M198 133c102 0 207 -45 207 -133c0 -92 -98 -133 -207 -133c-101 0 -198 46 -198 133c0 88 93 133 198 133zM293 -21c0 14 -3 29 -8 44c-7 20 -18 38 -33 54c-20 21 -43 31 -68 31l-20 -2c-15 -5 -27 -14 -36 -28c-4 -9 -6 -17 -8 -24s-3 -16 -3 -27c0 -15 3 -34 9 -57 s18 -41 34 -55c15 -15 36 -23 62 -23c4 0 10 1 18 2c19 5 32 15 40 30s13 34 13 55z" />
		</symbol>
	</defs>
	<style type="text/css">g.page-margin{font-family:Times;} g.tempo{font-weight:bold;} g.dir, g.dynam {font-style:italic;}</style>
	<svg id="definition-scale" viewBox="0 0 7920 3730">
		<g class="page-margin" transform="translate(500, 500)">
			<g class="system" id="system-0000001459809759">
				<g class="section boundaryStart" id="section-0000001175395249" />
				<g class="measure" id="measure-0000000165881190">
					<g class="staff" id="staff-0000000349825458">
						<path d="M0 1612 L3043 1612" stroke="currentColor" stroke-width="20" />
						<path d="M0 1792 L3043 1792" stroke="currentColor" stroke-width="20" />
						<path d="M0 1972 L3043 1972" stroke="currentColor" stroke-width="20" />
						<path d="M0 2152 L3043 2152" stroke="currentColor" stroke-width="20" />
						<path d="M0 2332 L3043 2332" stroke="currentColor" stroke-width="20" />
						<g class="clef" id="clef-0000000546677535">
							<use xlink:href="#E050" x="90" y="2152" height="720px" width="720px" />
						</g>
						<g class="meterSig" id="msig-0000002084050317" fill="currentColor" >
							<use xlink:href="#E084" x="735" y="1792" height="720px" width="720px" />
							<use xlink:href="#E084" x="735" y="2152" height="720px" width="720px" />
						</g>
						<g class="ledgerLines below">
							<path d="M1233 2512 L1623 2512" stroke="currentColor" stroke-width="35" />
						</g>
						<g class="layer" id="layer-0000001853730767">
							<g class="note" id="note-0000002107733940">
								<use xlink:href="#E0A2" x="1283" y="2512" height="720px" width="720px" />
							</g>
						</g>
					</g>
					<g class="dir" id="dir-0000000991882040">
						<text x="1428" y="1479" font-size="0px">
							<tspan font-size="405px" class="text">Allegro </tspan>
							<tspan class="rend" id="rend-0000001375531017">
								<tspan font-size="405px" class="text">molto</tspan>
							</tspan>
						</text>
					</g>
					<g class="barLineAttr" id="bline-0000000178519931">
						<path d="M3028 1612 L3028 2332" stroke="currentColor" stroke-width="30" />
					</g>
				</g>
				<g class="measure" id="measure-0000001070520604">
					<g class="staff" id="staff-0000001441475186">
						<path d="M3043 1612 L4983 1612" stroke="currentColor" stroke-width="20" />
						<path d="M3043 1792 L4983 1792" stroke="currentColor" stroke-width="20" />
						<path d="M3043 1972 L4983 1972" stroke="currentColor" stroke-width="20" />
						<path d="M3043 2152 L4983 2152" stroke="currentColor" stroke-width="20" />
						<path d="M3043 2332 L4983 2332" stroke="currentColor" stroke-width="20" />
						<g class="layer" id="layer-0000001110429295">
							<g class="note" id="note-0000001352268635">
								<use xlink:href="#E0A2" x="3223" y="2422" height="720px" width="720px" />
							</g>
						</g>
					</g>
					<g class="dir" id="dir-0000000479487140" fill="currentColor" >
						<text x="3368" y="1150" font-size="0px">
							<tspan class="rend" id="rend-0000001381718436" xml:space="preserve">
								<tspan font-size="405px" class="text">a         b       c</tspan>
							</tspan>
						</text>
					</g>
					<g class="barLineAttr" id="bline-0000001580895410">
						<path d="M4968 1612 L4968 2332" stroke="currentColor" stroke-width="30" />
					</g>
				</g>
				<g class="measure" id="measure-0000001950076222">
					<g class="staff" id="staff-0000001529093010">
						<path d="M4983 1612 L6923 1612" stroke="currentColor" stroke-width="20" />
						<path d="M4983 1792 L6923 1792" stroke="currentColor" stroke-width="20" />
						<path d="M4983 1972 L6923 1972" stroke="currentColor" stroke-width="20" />
						<path d="M4983 2152 L6923 2152" stroke="currentColor" stroke-width="20" />
						<path d="M4983 2332 L6923 2332" stroke="currentColor" stroke-width="20" />
						<g class="layer" id="layer-0000000529415421">
							<g class="note" id="note-0000000860231226">
								<use xlink:href="#E0A2" x="5163" y="2332" height="720px" width="720px" />
							</g>
						</g>
					</g>
					<g class="dir" id="dir-0000001631495210" fill="currentColor" >
						<text x="5308" y="822" font-size="0px">
							<tspan class="rend" id="rend-0000001468789574" xml:space="default">
								<tspan font-size="405px" class="text">a      b      c</tspan>
							</tspan>
						</text>
					</g>
					<g fill="currentColor" class="barLineAttr" id="bline-0000000686488442">
						<path d="M6908 1612 L6908 2332" stroke="currentColor" stroke-width="30" />
					</g>
				</g>
				<g class="boundaryEnd section-0000001175395249" id="bdend-0000000628458526" />
			</g>
		</g>
	</svg>
</svg>
```

**Comment by craigsapp** [comment]
Loading the second example (with the fills set to currentColor) into Inkscape:

![screen shot 2017-04-22 at 11 47 11 am](https://cloud.githubusercontent.com/assets/3487289/25307195/7a1fb892-2751-11e7-942a-56201d143ec3.png)

Probably a font substitution since "Allegro molto" loses its space.

**Comment by craigsapp** [comment]
Loading into Adobe Illustrator CS3:

<img width="859" alt="screen shot 2017-04-22 at 11 50 30 am" src="https://cloud.githubusercontent.com/assets/3487289/25307222/eed32606-2751-11e7-942d-8a38db0ce469.png">

**Comment by craigsapp** [comment]
Gnome bug report about a similar problem dealing with currentColor in SVGs being rendered with 0% opacity:
https://bugzilla.gnome.org/show_bug.cgi?id=764808

**Comment by rettinghaus** [comment]
Good that it is fixed in GNOME at last. But latest GIMP still shows this behavior. 
I'd recommend to wait for that feature some months.

**Comment by craigsapp** [comment]
It would be nice to have but low priority to make the SVG play nice with GIMP.  Also it drops to very low priority, because I forgot about the following solution that I use in VHV for highlighting notes (fills and strokes) on hover:

```css

g[id^="bline-"]:hover,
g[id^="clef-"]:hover,
g[id^="ksig-"]:hover,
g[id^="msig-"]:hover,
g[id^="note-"]:hover,
g[id^="accid-"]:hover,
g[id^="trill-"]:hover,
g[id^="artic-"]:hover,
g[id^="slur-"]:hover,
g[id^="tie-"]:hover,
g[id^="slur-"]:hover,
g[id^="fermata-"]:hover,
g[id^="rest-"]:hover {
        fill: orange;
}

g[id^="bline-"]:hover [stroke],
g[id^="clef-"]:hover [stroke],
g[id^="ksig-"]:hover [stroke],
g[id^="msig-"]:hover [stroke],
g[id^="note-"]:hover [stroke],
g[id^="accid-"]:hover [stroke],
g[id^="trill-"]:hover [stroke],
g[id^="artic-"]:hover [stroke],
g[id^="slur-"]:hover [stroke],
g[id^="tie-"]:hover [stroke],
g[id^="slur-"]:hover [stroke],
g[id^="fermta-"]:hover [stroke],
g[id^="rest-"]:hover [stroke] {
        stroke: orange;
        stroke-width: 50px;
}
```

Addressing the *stroke* attribute explicitly in CSS seems to be required in order to change it in SVG content.

Result of moving mouse anywhere over eighth note:

<img width="579" alt="screen shot 2017-04-22 at 12 27 54 pm" src="https://cloud.githubusercontent.com/assets/3487289/25307529/3ec26c30-2757-11e7-8ddb-3358c939a3ad.png">


Coloring everything a certain color:

```
svg g [stroke] {
        stroke: hotpink;
}

svg g {
        fill: hotpink;
}

svg g [stroke]:hover {
        stroke: lawngreen;
}

svg g:hover {
        fill: lawngreen;
}
```


<img width="581" alt="screen shot 2017-04-22 at 12 36 12 pm" src="https://cloud.githubusercontent.com/assets/3487289/25307599/9f67e852-2758-11e7-887d-ceae3bcca0cb.png">

**Comment by craigsapp** [comment]
Colorful music:

<img width="563" alt="screen shot 2017-04-22 at 2 35 08 pm" src="https://cloud.githubusercontent.com/assets/3487289/25308442/13968624-2769-11e7-86ae-84e4fdb33f9c.png">
